### ISSUE TITLE
ApiClient . setKeyManagers() is not called after #175

### ISSUE THREAD

**Issue by ceeaspb** [kubernetes-client/java#197]
Rebasing to commit the couple of fixes I found the ApiKey config no longer works.

dbee39eb0cff36f5bf2679ed894fbbe4a41b8391 PR #175 seems to have broken it, previous commit works.

I'm running a client that is reading the config from file during dev, not running inside a k8s cluster.

```testApiKeyConfigbuilder``` looks like it's testing the code.
```+               assertEquals( apiKey, ((io.kubernetes.client.auth.ApiKeyAuth)client.getAuthentications().get("BearerToken")).getApiKey());```

passes but as far as I could tell from stepping through the latest broken code the setApiKey() method is never called so for a real application the BearerToken is never set resulting in 403 Forbidden.

**Comment by kondapally1989** [comment]
@ceeaspb  io.kubernetes.client.util.credentials.AccessTokenAuthentication.provide() does call setApiKey();

when client is built  on token
```
ClientBuilder()
        .setBasePath(url)
        .setAuthentication(new AccessTokenAuthentication(token))
        .setVerifyingSsl(validateSSL)
        .build()
```
or 
```
Config.fromToken(url, token);
```

**Comment by ceeaspb** [comment]
Running the 1st example from the README page fails (for me at least).

* example running from the command line on my laptop
* against a minikube environment
* the example 

```
    public static void main(String[] args) throws IOException, ApiException{
        ApiClient client = Config.defaultClient();
        Configuration.setDefaultApiClient(client);

        CoreV1Api api = new CoreV1Api();
        V1PodList list = api.listPodForAllNamespaces(null, null, null, null, null, null, null, null, null);
        for (V1Pod item : list.getItems()) {
            System.out.println(item.getMetadata().getName());
        }
    }
```

fails with

```
Exception in thread "main" io.kubernetes.client.ApiException: Forbidden
	at io.kubernetes.client.ApiClient.handleResponse(ApiClient.java:882)
	at io.kubernetes.client.ApiClient.execute(ApiClient.java:798)
	at io.kubernetes.client.apis.CoreV1Api.listPodForAllNamespacesWithHttpInfo(CoreV1Api.java:18462)
	at io.kubernetes.client.apis.CoreV1Api.listPodForAllNamespaces(CoreV1Api.java:18440)
	at K8sScratch.main(K8sScratch.java:36)
```

**Comment by ceeaspb** [comment]
ApiClient . setKeyManagers() is not called.

**Comment by ceeaspb** [comment]
looks like a problem with client cert :

working:
```
*** ServerHelloDone
[read] MD5 and SHA1 hashes:  len = 4
0000: 0E 00 00 00                                        ....
matching alias: cn=minikube-user,o=system:masters
*** Certificate chain
```

not working

```
*** ServerHelloDone
[read] MD5 and SHA1 hashes:  len = 4
0000: 0E 00 00 00                                        ....
Warning: no suitable certificate found - continuing without client authentication
*** Certificate chain
```

**Comment by ceeaspb** [comment]
@brendanburns @lewisheadden would this fix it?
We no longer appear to call provide on ClientCertificateAuthentication, also the ordering of the cert set up got muddled.

cf. comment
```
-            // It's silly to have to do it in this order, but each SSL setup
-            // consumes the CA cert, so if we do this before the client certs
-            // are injected the cert input stream is exhausted and things get
-            // grumpy'
```

```
 git status
HEAD detached at dbee39e
```

```
diff --git a/kubernetes/src/main/java/io/kubernetes/client/ApiClient.java b/kubernetes/src/main/java/io/kubernetes/client/ApiClient.java
index c6bccd3..985d0ca 100644
--- a/kubernetes/src/main/java/io/kubernetes/client/ApiClient.java
+++ b/kubernetes/src/main/java/io/kubernetes/client/ApiClient.java
@@ -196,7 +196,6 @@ public class ApiClient {
      */
     public ApiClient setSslCaCert(InputStream sslCaCert) {
         this.sslCaCert = sslCaCert;
-        applySslSettings();
         return this;
     }
 
diff --git a/target/checkout b/target/checkout
deleted file mode 160000
index 02bb7ab..0000000
--- a/target/checkout
+++ /dev/null
@@ -1 +0,0 @@
-Subproject commit 02bb7ab2ee7495b2a64148e9033b2cb9777f68e2
diff --git a/util/src/main/java/io/kubernetes/client/util/credentials/KubeconfigAuthentication.java b/util/src/main/java/io/kubernetes/client/util/credentials/KubeconfigAuthentication.java
index d208058..12e54c6 100644
--- a/util/src/main/java/io/kubernetes/client/util/credentials/KubeconfigAuthentication.java
+++ b/util/src/main/java/io/kubernetes/client/util/credentials/KubeconfigAuthentication.java
@@ -32,7 +32,7 @@ public class KubeconfigAuthentication implements Authentication {
 
   @Override public void provide(ApiClient client) {
     if(clientCert != null && clientKey != null) {
-      new ClientCertificateAuthentication(clientCert, clientKey);
+      new ClientCertificateAuthentication(clientCert, clientKey).provide(client);
     }
 
     if(username != null && password != null) {
```

**Comment by brendandburns** [comment]
I see the call to `provide()` in the code now here:

https://github.com/kubernetes-client/java/blob/master/util/src/main/java/io/kubernetes/client/util/credentials/KubeconfigAuthentication.java#L39

I just tried from HEAD with #199 applied and it worked correctly for me with an external cert-based auth. I can try with a Minikube environment later today...

**Comment by ceeaspb** [comment]
My apologies @brendanburns @kondapally1989 it looks like I had some problem pulling latest from the repo as I did not have that revision locally, and then went down a rabbit whole looking at the diff of PR 175.
I spotted the double call to applySslSettings() as part of this which I guess is #199, and now see the resulting error to that with minikube. Once that is fixed I can commit the BearerToken change.
